### DISCUSSION TITLE
Disable Output Window [SOLVED]

### DISCUSSION THREAD

**Discussion by jasc2v8** [mark-wiemer/ahkpp#67] [General]
Disable the [annoying] Output Window by adding the following code to out.js:   😃 

C:\Users\USER\.vscode\extensions\mark-wiemer.vscode-autohotkey-plus-plus-2.7.1\out\common\out.js

```
class Out {
    static log(value) {
			
	//disable output window by adding this return:
	return
				
        if (!this.channel) {
            this.channel = vscode.window.createOutputChannel('AHK');
        }
        this.channel.show(true);
        this.channel.appendLine(value + '');
    }
}
```

By the way Mark, you've done a **FANTASTIC** job on this extension - **THANK YOU!**
-jasc2v8

**Comment by fade2gray** [comment]
> Output Window

Would that be the `IntelliSense` output, i.e. #48?

**Reply by jasc2v8** [reply]
No, but I'll post another topic about that.
See the post below for Output Window explanation.

**Comment by FuPeiJiang** [comment]
@jasc2v8
what is the "[annoying] Output Window" ?
can you send image ?

**Reply by jasc2v8** [reply]
The Output window displays status messages for various features in the integrated development environment (IDE).
To open the Output window, on the menu bar, choose View > Output, or press Ctrl+Alt+O.
On my system, when I run an AHK script and I get an error (happens often...), the Output window opens.
I have to manually close the window to get it out of the way, or press Ctrl-J (awkard).
Now if my script fails I don't see the output window.
I can always open it if I need to.

**Reply by FuPeiJiang** [reply]
I have an ahk script to run ahk script
I set the title of vscode to the path of active file
in settings.json:
`"window.title": "${activeEditorLong} - Visual Studio Code",`
now ahk can see it.
I press f5 to save (send ^s), close the script, run the script

I close the script because `#SingleInstance, force` doesn't always work
___
I see no point in using the extension to run the script

**Comment by mark-wiemer** [comment]
I appreciate the praise, but of course most of the credit goes to `cweijan` who extended this extension's functionality greatly from `stef-levesque` and others who contributed in the early days. I'm primarily focused on bugfixing :)

If you'd like me to add this "Disable output window" functionality as a setting in the extension, feel free to open an issue for it
